src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -666,14 +666,26 @@ private BoundExpression BindTupleExpression(TupleExpressionSyntax node, Diagnost
 
             var boundArguments = ArrayBuilder<BoundExpression>.GetInstance(arguments.Count);
             var elementTypes = ArrayBuilder<TypeSymbol>.GetInstance(arguments.Count);
+            var elementLocations = ArrayBuilder<Location>.GetInstance(arguments.Count);
             ArrayBuilder<string> elementNames = null;
             int countOfExplicitNames = 0;
 
             // prepare and check element names and types
             for (int i = 0, l = numElements; i < l; i++)
             {
                 ArgumentSyntax argumentSyntax = arguments[i];
-                string name = argumentSyntax.NameColon?.Name?.Identifier.ValueText;
+                string name = null;
+                IdentifierNameSyntax nameSyntax = argumentSyntax.NameColon?.Name;
+
+                if (nameSyntax != null)
+                {
+                    name = nameSyntax.Identifier.ValueText;
+                    elementLocations.Add(nameSyntax.Location);
+                }
```

Comment:
`CheckTupleMemberName` could be called in this branch, rather than in separate check below. This would be similar to the change below, in `BindTupleType`.